lang/cs/Org.Apache.REEF.IMRU/API/UpdateResult.cs

```diff
@@ -50,22 +50,34 @@ internal bool IsNotDone
             get { return !_done; }
         }
 
-        internal bool HasMapInput
+        /// <summary>
+        /// Indicates if the UpdateResult has map input data
+        /// </summary>
+        public bool HasMapInput
```

Comment:
@dkm2110 Have you considered the approach I've suggested? If this PR can be merged to unblock your work on state preservation in the algorithm, we can work on hiding properties again in parallel